addons/mail/static/tests/discuss/core/public_web/sub_channels.test.js

```diff
@@ -192,3 +193,75 @@ test("mention suggestions in thread match channel restrictions", async () => {
+    await click("button[aria-label='Create Thread']");
+    await contains(".o-mail-Discuss-threadName", { value: "New Thread" });
+    await contains(".o-mail-DiscussSidebarChannel-subChannel.opacity-50", { count: 0 });
+    await click("button[title='Notification Settings']");
+    await click("button:contains('Mute Conversation')");
+    await click("button:contains('Until I turn it back on')");
+    await contains(".o-mail-DiscussSidebarChannel-subChannel.opacity-50", { count: 1 });
+});
+
+test("sub channel is not shown when parent channel is muted", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Discuss-threadName", { value: "General" });
+    await click("button[title='Threads']");
+    await click("button[aria-label='Create Thread']");
+    await contains(".o-mail-Discuss-threadName", { value: "New Thread" });
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 1 });
+    await click(".o-mail-DiscussSidebarChannel");
+    await click("button[title='Notification Settings']");
+    await click("button:contains('Mute Conversation')");
+    await click("button:contains('Until I turn it back on')");
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 0 });
+});
+
+test("sub channel is shown when parent channel is muted but sub channel has unread messages", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    const partnerId2 = pyEnv["res.partner"].create(
+        {
+            email: "p1@odoo.com", name: "p1",
+        },
+    )
+    const userId2 = pyEnv["res.users"].create({ name: "User 2", partner_id: partnerId2 });
+    const partnerId = serverState.partnerId;
+    const subChannelId = pyEnv["discuss.channel"].create({
+        name: "Thread",
+        parent_channel_id: channelId,
+        channel_member_ids: [
+            Command.create({ partner_id: partnerId }),
+            Command.create({ partner_id: partnerId2 })
+        ],
+    });
+    await start();
+    await openDiscuss(channelId);
+    await click(".o-mail-DiscussSidebarChannel");
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 1 });
+    await click("button[title='Notification Settings']");
+    await click("button:contains('Mute Conversation')");
+    await click("button:contains('Until I turn it back on')");
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 0 });
+    withUser(userId2, () =>
+        rpc(
+            "/mail/message/post",
+            {
+                post_data: { body: "Some message", message_type: "comment" },
+                thread_id: subChannelId,
+                thread_model: "discuss.channel",
+            }
+        )
+    )
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 1 });
```

Comment:
Can remove the `count: 1`: this is the default mode. Code looks easier without the explicit `1`.
If there are more than 1, this will crash and tell how many it found instead.